File: echo-guardian-production-foundation-v1.0/native/ios/EchoGuardianKit/Sources/EchoGuardianKit/Status/StatusViewModel.swift
import Foundation

public struct PlainStatus: Codable, Equatable {
    public let monitoringOn: Bool
    public let activeSpaces: [String]
    public let activeSignalTypes: [String]
    public let authorityLabel: String
    public let degraded: Bool
    public let whatAreYouDoing: String
}

public final class StatusViewModel {
    public init() {}

    public func makeStatus(monitoringOn: Bool, spaces: [String], signals: [String], authority: String, degraded: Bool) -> PlainStatus {
        let onOff = monitoringOn ? "on" : "off"
        let spacesText = spaces.isEmpty ? "no rooms" : spaces.joined(separator: ", ")
        let signalText = signals.isEmpty ? "no safety signals" : signals.joined(separator: ", ")
        let degradedText = degraded ? " Some parts need attention." : " Everything needed for this local status view is working."
        return PlainStatus(
            monitoringOn: monitoringOn,
            activeSpaces: spaces,
            activeSignalTypes: signals,
            authorityLabel: authority,
            degraded: degraded,
            whatAreYouDoing: "Echo Guardian monitoring is \(onOff). Active spaces: \(spacesText). It uses: \(signalText). Control: \(authority).\(degradedText)"
        )
    }
}

File: echo-guardian-production-foundation-v1.0/native/android/echo-guardian-core/src/main/java/org/echoguardian/status/StatusViewModel.kt
package org.echoguardian.status

data class PlainStatus(
    val monitoringOn: Boolean,
    val activeSpaces: List<String>,
    val activeSignalTypes: List<String>,
    val authorityLabel: String,
    val degraded: Boolean,
    val whatAreYouDoing: String
)

class StatusViewModel {
    fun makeStatus(monitoringOn: Boolean, spaces: List<String>, signals: List<String>, authority: String, degraded: Boolean): PlainStatus {
        val onOff = if (monitoringOn) "on" else "off"
        val spacesText = if (spaces.isEmpty()) "no rooms" else spaces.joinToString(", ")
        val signalText = if (signals.isEmpty()) "no safety signals" else signals.joinToString(", ")
        val degradedText = if (degraded) " Some parts need attention." else " Everything needed for this local status view is working."
        return PlainStatus(monitoringOn, spaces, signals, authority, degraded, "Echo Guardian monitoring is $onOff. Active spaces: $spacesText. It uses: $signalText. Control: $authority.$degradedText")
    }
}
